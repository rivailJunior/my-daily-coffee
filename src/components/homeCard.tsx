import Image from 'next/image';
import { Button } from './ui/button';
import { Link } from 'lucide-react';

interface HomeCardProps {
  onClick?: () => void;
  title: string;
  description: string;
  image: string;
}

export const HomeCard = ({
  onClick,
  title,
  description,
  image,
}: HomeCardProps) => {
  return (
    <div className=' bg-white border border-gray-200 rounded-lg shadow-sm dark:bg-gray-800 dark:border-gray-700'>
      <Image
        className='rounded-t-lg w-full h-48 md:h-80 object-cover'
        src={image}
        alt=''
        width={500}
        height={500}
      />
      <div className='p-5'>
        <a href='#'>
          <h5 className='mb-2 text-2xl font-bold tracking-tight text-gray-900 dark:text-white'>
            {title}
          </h5>
        </a>
        <p className='mb-3 font-normal text-gray-700 dark:text-gray-400'>
          {description}
        </p>
        <Button variant='default' onClick={onClick}>
          See more
        </Button>
      </div>
    </div>
  );
};

export const HomeCardHorizontal = ({
  title,
  description,
  image,
  createdAt,
}: HomeCardProps & { createdAt: string }) => {
  return (
    <div className='flex flex-row bg-white border border-gray-200 rounded-lg overflow-hidden shadow-sm md:flex-row w-full hover:bg-gray-100 dark:border-gray-700 dark:bg-gray-800 dark:hover:bg-gray-700 h-auto'>
      <Image
        className='object-cover max-w-20 h-auto md:h-auto md:max-w-24 md:rounded-none md:rounded-s-lg'
        src={image}
        alt=''
        width={500}
        height={500}
      />
      <div className='flex flex-col  p-4 leading-normal'>
        <div className='text-lg font-medium tracking-tight text-gray-900 dark:text-white capitalize'>
          {title}
        </div>
        <div className='text-md text-gray-700 dark:text-gray-400 pt-1 capitalize'>
          {description}
        </div>
        <div className='text-sm text-gray-700 dark:text-gray-400 pt-4'>
          {createdAt}
        </div>
      </div>
    </div>
  );
};