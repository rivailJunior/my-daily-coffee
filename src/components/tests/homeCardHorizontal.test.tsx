import { render, screen } from '@testing-library/react';
import { HomeCardHorizontal } from '../homeCard';

beforeEach(() => {
  jest.clearAllMocks();
});

describe('HomeCardHorizontal', () => {
  const defaultProps = {
    title: 'Horizontal Test Title',
    description: 'Horizontal Test Description',
    image: '/test-horizontal-image.jpg',
    createdAt: '2023-01-01',
  };

  it('renders with required props', () => {
    render(<HomeCardHorizontal {...defaultProps} />);

    expect(screen.getByText('Horizontal Test Title')).toBeInTheDocument();
    expect(screen.getByText('Horizontal Test Description')).toBeInTheDocument();
    expect(screen.getByRole('img')).toHaveAttribute('src');
    expect(screen.getByText('2023-01-01')).toBeInTheDocument();
  });

  it('applies horizontal layout styles', () => {
    render(<HomeCardHorizontal {...defaultProps} />);

    const card = screen.getByText('Horizontal Test Title').closest('div');
    expect(card).toHaveClass(
      'text-lg font-medium tracking-tight text-gray-900 dark:text-white capitalize'
    );
  });

  it('matches snapshot', () => {
    const { container } = render(<HomeCardHorizontal {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });

  it('capitalizes the title', () => {
    render(<HomeCardHorizontal {...defaultProps} title='test title' />);

    const titleElement = screen.getByText('test title');
    expect(titleElement).toHaveClass('capitalize');
  });
});
