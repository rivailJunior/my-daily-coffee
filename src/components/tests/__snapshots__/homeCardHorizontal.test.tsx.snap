// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeCardHorizontal matches snapshot 1`] = `
<div>
  <div
    class="flex flex-row bg-white border border-gray-200 rounded-lg overflow-hidden shadow-sm md:flex-row w-full hover:bg-gray-100 dark:border-gray-700 dark:bg-gray-800 dark:hover:bg-gray-700 h-auto"
  >
    <img
      alt=""
      class="object-cover max-w-20 h-auto md:h-auto md:max-w-24 md:rounded-none md:rounded-s-lg"
      data-nimg="1"
      decoding="async"
      height="500"
      loading="lazy"
      src="/_next/image?url=%2Ftest-horizontal-image.jpg&w=1080&q=75"
      srcset="/_next/image?url=%2Ftest-horizontal-image.jpg&w=640&q=75 1x, /_next/image?url=%2Ftest-horizontal-image.jpg&w=1080&q=75 2x"
      style="color: transparent;"
      width="500"
    />
    <div
      class="flex flex-col  p-4 leading-normal"
    >
      <div
        class="text-lg font-medium tracking-tight text-gray-900 dark:text-white capitalize"
      >
        Horizontal Test Title
      </div>
      <div
        class="text-md text-gray-700 dark:text-gray-400 pt-1 capitalize"
      >
        Horizontal Test Description
      </div>
      <div
        class="text-sm text-gray-700 dark:text-gray-400 pt-4"
      >
        2023-01-01
      </div>
    </div>
  </div>
</div>
`;
